src/box/lua/execute.c

```diff
@@ -411,12 +401,8 @@ lua_sql_bind_decode(struct lua_State *L, struct sql_bind *bind, int idx, int i)
 			return -1;
 		}
 		bind->bytes = region_used(region) - used;
-		bind->s = region_join(region, bind->bytes);
-		if (bind->s != NULL)
-			break;
-		region_truncate(region, used);
-		diag_set(OutOfMemory, bind->bytes, "region_join", "bind->s");
-		return -1;
+		bind->s = xregion_join(region, bind->bytes);
```

Comment:
There's `OutOfMemory` above set when `region_serve_cb` or `region_alloc_cb` fails. Can you remove that too in the scope of this work? Same comment to other functions using `mpstream`, for example `sql_encode_table`.